src/service/router.ts

```diff
@@ -2,21 +2,104 @@ import { errorHandler } from '@backstage/backend-common';
 import express from 'express';
 import Router from 'express-promise-router';
 import { Logger } from 'winston';
+import { Config } from '@backstage/config';
+import { PluginDatabaseManager } from '@backstage/backend-common';
+import { resolvePackagePath } from '@backstage/backend-common';
+import { Knex } from 'knex'
+import { getReposData, graphqlInput, ValidRepo } from './pull_request_query';
+import { graphql } from '@octokit/graphql'
 
 export interface RouterOptions {
   logger: Logger;
+  config: Config;
+  database: PluginDatabaseManager;
+}
+
+async function applyDatabaseMigrations(knex: Knex): Promise<void> {
+  const migrationsDir = resolvePackagePath(
+    '@internal/pr-tracker-backend',
+    'migrations'
+  );
+
+  await knex.migrate.up({
+    directory: migrationsDir,
+  });
 }
 
 export async function createRouter(options: RouterOptions): Promise<express.Router> {
-  const { logger } = options;
+  const { logger, config, database } = options;
+  const databaseClient = await database.getClient();
+  await applyDatabaseMigrations(databaseClient);
```

Comment:
Do this to fix the type issue
```suggestion
  await applyDatabaseMigrations(databaseClient as unknown as Knex);
```

Reply:
I've changed it but I didn't see a type issue...